pipelines/rhtas-operator-e2e.yaml

```diff
@@ -300,10 +310,59 @@ spec:
                 mountPath: /repository
               - name: binaries
                 mountPath: /binaries
+              - name: dump
+                mountPath: /dump
             script: |
+              #!/bin/sh
+              set +e
               cd /repository/source
               export PATH="$PATH:/binaries"
               export OIDC_ISSUER_URL=https://$OIDC_HOST/auth/realms/trusted-artifact-signer
               openssl s_client -connect $OIDC_HOST:443 > /tmp/ssl.cert
               export SSL_CERT_FILE=/tmp/ssl.cert
               make test-e2e
+              
+              if [ $? -eq 0 ]; then
+                echo -n "success" > "$(step.results.status.path)"
+              else
+                cp test/**/k8s-dump-*.tar.gz /dump
+                echo "Dumping files:"
+                ls /dump
+                echo -n "fail" >  "$(step.results.status.path)"
+                exit 1
+              fi
+          - name: secure-push-oci
+            when:
+              - input: "$(steps.execute-e2e.results.status)"
+                operator: notin
+                values: [ "success" ]
```

Comment:
Is it possible to define a `when:` blocks here like this, in a step definition, maybe it is different for stepActions? Im not too familiar with them.

Reply:
I did not get your point. Can you elaborate more?

Reply:
No real point, :), I was just unaware we could define `when:` blocks in step definitions. thats all :)